src/main/java/org/generationcp/middleware/operation/builder/LotBuilder.java

```diff
@@ -78,33 +86,39 @@ public LotBuilder(HibernateSessionProvider sessionProviderForLocal) {
 		return newGids;
 	}
 
-	private List<Lot> createLotsForAdd(List<Integer> gids, Integer locationId, Integer scaleId, String comment, Integer userId,
-			Double amount, Integer sourceId) throws MiddlewareQueryException {
-		List<Lot> lots = new ArrayList<>();
+	private List<Lot> createLotsForAdd(
+		final List<Integer> gids, final Integer locationId, final Integer scaleId, final String comment, final Integer userId,
+		final Double amount, final Integer sourceId, final CropType cropType) throws MiddlewareQueryException {
+		final List<Lot> lots = new ArrayList<>();
 
 		if (gids != null && !gids.isEmpty()) {
-			for (Integer gid : gids) {
+			for (final Integer gid : gids) {
 
-				lots.add(this.createLotForAdd(gid, locationId, scaleId, comment, userId));
+				lots.add(this.createLotForAdd(gid, locationId, scaleId, comment, userId, cropType));
 			}
 		}
 
 		return lots;
 	}
 
-	public Lot createLotForAdd(Integer gid, Integer locationId, Integer scaleId, String comment, Integer userId)
+	public Lot createLotForAdd(
+		final Integer gid, final Integer locationId, final Integer scaleId, final String comment, final Integer userId,
+		final CropType cropType)
 			throws MiddlewareQueryException {
-		return new Lot(null, userId, EntityType.GERMPLSM.name(), gid, locationId, scaleId, LotStatus.ACTIVE.getIntValue(),
-				LotBuilder.LOT_NOT_DERIVED_FROM_ANOTHER, comment);
+		final Lot lot = new Lot(null, userId, EntityType.GERMPLSM.name(), gid, locationId, scaleId, LotStatus.ACTIVE.getIntValue(),
+			LotBuilder.LOT_NOT_DERIVED_FROM_ANOTHER, comment);
+		this.lotIDGenerator = new LotIDGeneratorImpl();
```

Comment:
@vmaletta lotIDGenerator should be @Autowired, remove manually construction

Reply:
As we removed lotIdGenerator we dont need autowired